webapp/src/components/Variants/Edition/index.tsx

```diff
@@ -121,6 +134,43 @@ const EditionView = (props: PropTypes) => {
     setCommands(tmpCommand);
   };
 
+  const onCommandImport = (index: number, json: object) => {
+    let tmpCommand: Array<CommandItem> = [];
+    tmpCommand = tmpCommand.concat(commands);
+    // eslint-disable-next-line dot-notation
+    tmpCommand[index].action = (json as any)['action'];
+    // eslint-disable-next-line dot-notation
+    tmpCommand[index].args = { ...((json as any)['args'] as object) };
+    tmpCommand[index].updated = true;
+    setCommands(tmpCommand);
+    enqueueSnackbar(t('variants:importSuccess'), { variant: 'success' });
+  };
+
+  const onCommandExport = (index: number) => {
+    const elm = commands[index];
+    exportJson({ action: elm.action, args: elm.args }, `${elm.id}_command.json`);
+  };
+
+  const onGlobalExport = () => {
+    exportJson(fromCommandItemToJsonCommand(commands), `${studyId}_commands.json`);
+  };
+
+  const onGlobalImport = async (json: object) => {
+    try {
+      const globalJson: Array<JsonCommandItem> = (json as Array<JsonCommandItem>);
+
+      await deleteAllCommands(studyId);
+
+      await appendCommands(studyId, globalJson);
```

Comment:
as discussed, you have to combine those two calls in one.
We want to use the back to check the command validity.
Here, if I upload a proper json containing wrong commands (with wrong action names or args), it will delete all previous commands and apply the new wrong ones or just fail if the json doesn't contains commands.

Reply:
or you could at least call `checkCommandValidity` on the globalJson to check if at least the content look like commands.

Reply:
so, in appendCommand, appendCommands, updateCommand and a newly replaceCommands, it should check command validity by calling something like `self.command_factory.to_icommand()` which will try to read a real ICommand with validation from the command dtos.